js/data/custom_store.d.ts

```diff
@@ -38,10 +39,10 @@ export interface CustomStoreOptions<
     /**
      * @docid
      * @type_function_param1 options:LoadOptions
-     * @type_function_return Promise<any>|Array<any>
+     * @type_function_return Promise<Array>|Array
      * @public
      */
-    load: ((options: LoadOptions<TItem>) => PromiseLike<TItem> | Array<TItem>);
+    load: ((options: LoadOptions<TItem>) => DxPromise<Array<TItem>> | Array<TItem>);
```

Comment:
It is possible to return promise with { data: Array<TItem>, totalCount: number, summary: Array} object here.
Also grouped data can be returned.

Reply:
See details examples for jquery and angular here - https://js.devexpress.com/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/